lib/sdam/topology.js

```diff
@@ -241,6 +264,55 @@ class Topology extends EventEmitter {
     });
   }
 
+  /**
+   * Internal method. ignore.
+   * @ignore
+   */
+  handleSrvPolling(ev) {
+    const newAddresses = ev.addresses();
+    const serversToRemove = new Set();
+    for (const server of this.s.description.servers) {
```

Comment:
Should this logic be moved into `TopologyDescription.update`? That method has a lot of replica set specific functions like `updateRsFromPrimary`, that do the set differences necessary to grow/shrink the known set of server descriptions.

I think we just need to think of SRV polling as "replica set behavior for mongoses", so basically we could create `ServerDescription`s for each of the incoming addresses, and add all of the other addresses in the `hosts` field of the server description. Then `TopologyDescription.update` can do the resolution, and we don't need to change any other behavior here (since this already knows how to add/delete servers based on topology description changes).

Reply:
I have an idea for this, but it may be a little complicated. I don't think it is as simple as adding it to `TopologyDescription.update`